cpp/oneapi/dal/detail/common.hpp

```diff
@@ -166,4 +167,35 @@ inline void check_mul_overflow(const Data& first, const Data& second) {
     integer_overflow_ops<Data>{}.check_mul_overflow(first, second);
 }
 
+template <typename Data>
+struct limits {
+    static constexpr Data min() {
+        return std::numeric_limits<Data>::min();
+    }
+    static constexpr Data max() {
+        return std::numeric_limits<Data>::max();
+    }
+};
+
+template <typename Out, typename In>
+inline Out integral_asserted_cast(const In& value) {
```

Comment:
Two remarks here:
1. I see a trend to have long names in the service functionality of oneDAL - I don't like it much. Let's make the names shorter when possible? My suggestion here is to rename this function to `assert_safe_cast()` - "integral" is useless here in my opinion. In the future, similar functions like `safe_cast()` or `check_safe_cast()` can be added.
2. Does the `ONEDAL_ASSERT` permit to declare this as `constexpr`?

Reply:
1. Contradicts with https://github.com/oneapi-src/oneDAL/pull/1209#discussion_r512862268